src/main/kotlin/com/hypto/iam/server/service/UsersService.kt

```diff
@@ -150,6 +152,22 @@ class UsersServiceImpl : KoinComponent, UsersService {
         val userHrn = ResourceHrn(organizationId, "", IamResources.USER, userName)
         return getUser(userHrn, user)
     }
+
+    override suspend fun authenticate(userName: String, password: String): User {
+        if (!appConfig.app.uniqueUsersAcrossOrganizations)
+            throw BadRequestException("Email not unique across organizations. " +
+                "Please use Token APIs with organization ID")
+
+        val userRecord = userRepo.findByEmail(userName)
+            ?: throw UserNotFoundException("Email not found, Unable to authenticate user")
```

Comment:
nit: Error message can be "Email or Password is not valid. Unable to authenticate"